hummingbird/ml/convert.py

```diff
@@ -164,33 +155,77 @@ def _convert_onnxml(model, backend, test_input, device, extra_config={}):
-            assert len(initial_types[0][1].shape) == 2, "Hummingbird currently support only inputs with len(shape) == 2."
-
             from onnxconverter_common.data_types import FloatTensorType, DoubleTensorType, Int32TensorType, Int64TensorType
 
-            test_input = np.random.rand(initial_types[0][1].shape[0], initial_types[0][1].shape[1])
+            if initial_types is None:
+                # Generate inputs from the ONNX schema.
+                initial_types = []
+                for input in model.graph.input:
+                    name = input.name if hasattr(input, "name") else None
+                    data_type = (
+                        input.type.tensor_type.elem_type
+                        if hasattr(input, "type")
+                        and hasattr(input.type, "tensor_type")
+                        and hasattr(input.type.tensor_type, "elem_type")
+                        else None
+                    )
+                    if name is None:
+                        raise RuntimeError(
+                            "Cannot fetch input name or data_type from the ONNX schema. Please provide some test input."
+                        )
+                    if data_type is None:
+                        raise RuntimeError(
+                            "Cannot fetch input data_type from the ONNX schema, or data type is not tensor_type. Please provide some test input."
+                        )
+                    if not hasattr(input.type.tensor_type, "shape"):
+                        raise RuntimeError("Cannot fetch input shape from ONNX schema. Please provide some test input.")
+                    shape = [dim.dim_value for dim in input.type.tensor_type.shape.dim]
+
+                    if data_type == 1:
+                        initial_types.append((name, FloatTensorType(shape)))
+                    elif data_type == 11:
+                        initial_types.append((name, DoubleTensorType(shape)))
+                    elif data_type == 6:
+                        initial_types.append((name, Int32TensorType(shape)))
+                    elif data_type == 7:
+                        initial_types.append((name, Int64TensorType(shape)))
+                    else:
+                        raise RuntimeError(
+                            "Input data type {} not supported. Please fill an issue at https://github.com/microsoft/hummingbird/.".format(
+                                data_type
+                            )
+                        )
+
+            assert all(
+                map(lambda x: len(x[1].shape) == 2, initial_types)
+            ), "Hummingbird currently support only inputs with len(shape) == 2."
+            first_shape = initial_types[0][1].shape
+            first_type = type(initial_types[0][1])
+            assert all(
+                map(lambda x: x[1].shape == first_shape, initial_types)
+            ), "Hummingbird currently support only inputs with same shape."
```

Comment:
`%s/support/supports/g`

Also, can you add the shapes they provided in the output? onnx error messages are always so obscure to search for. :)

Reply:
You mean checking that the output shape is correct? I am actually by default generating sym dimensions so hopefully we won't have that problem.